tools/main.py

```diff
@@ -47,17 +59,18 @@ def build_auxiliary_model(config_file, num_classes, use_gpu, device_ids=None, we
 
     model = torchreid.models.build_model(**model_kwargs(cfg, num_classes))
 
-    if check_isfile(weights):
+    if (weights is not None) and (check_isfile(weights)):
         load_pretrained_weights(model, weights)
 
     if cfg.use_gpu:
         assert device_ids is not None
 
         model = DataParallel(model, device_ids=device_ids, output_device=0).cuda(device_ids[0])
 
-    optimizer = torchreid.optim.build_optimizer(model, **optimizer_kwargs(cfg))
-    scheduler = torchreid.optim.build_lr_scheduler(optimizer, **lr_scheduler_kwargs(cfg))
-
+    op_builder = torchreid.optim.OtimizerBuilder(model, **optimizer_kwargs(cfg))
```

Comment:
Is it necessary to split creation of builder and instantiation of target object?